fluss-server/src/main/java/org/apache/fluss/server/coordinator/RemoteStorageCleaner.java

```diff
@@ -57,10 +61,16 @@ public RemoteStorageCleaner(Configuration configuration, ExecutorService ioExecu
         }
     }
 
-    public void asyncDeleteTableRemoteDir(TablePath tablePath, boolean isKvTable, long tableId) {
+    public void asyncDeleteTableRemoteDir(
+            TablePath tablePath, boolean isKvTable, boolean isLakeEnabled, long tableId) {
         if (isKvTable) {
             asyncDeleteDir(FlussPaths.remoteTableDir(remoteKvDir, tablePath, tableId));
         }
+        if (isLakeEnabled) {
```

Comment:
If a table was enabled datalake, but turned off later, and then the table was deleted. We may leave the lake snapshot metadata file here? Maybe we can simply do the delete for the metadata file, no matter it is datalake enabled or not? (add comment for this if we do such change)